adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,928 @@
+= SYCL_KHR_FREE_FUNCTION_COMMANDS
+
+This extension provides an alternative mechanism for submitting commands to a
+device via free-functions that require developers to opt-in to the creation of
+[code]#event# objects.
+
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_FREE_FUNCTION_COMMANDS# to one of the values defined in the
+table below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+== Usage example
+
+The example below rewrites the application from
+<<subsec:example.sycl.application>> to demonstrate the usage of this extension.
+
+[source,role=synopsis]
+----
+#include <iostream>
+#include <sycl/sycl.hpp>
+using namespace sycl; // (optional) avoids need for "sycl::" before SYCL names
+
+// Size of the matrices
+constexpr size_t N = 2000;
+constexpr size_t M = 3000;
+
+int main() {
+  // Create a queue to work on
+  queue myQueue;
+
+  // Create some 2D arrays of float for our matrices
+  float* a = malloc_shared<float>(N * M, myQueue);
+  float* b = malloc_shared<float>(N * M, myQueue);
+  float* c = malloc_shared<float>(N * M, myQueue);
+
+  // Launch an asynchronous kernel to initialize a
+  // Use khr::submit to get a handler, even though not required here
+  khr::submit(myQueue, [&](handler& cgh) {
+
+    // Enqueue a kernel iterating on a N*M 2D iteration space
+    khr::launch(cgh, range<2> { N, M }, [=](id<2> index) {
+      size_t i = index[0];
+      size_t j = index[1];
+      a[i * M + j] = i * 2 + j;
+    });
+
+  });
+
+  // Launch an asynchronous kernel to initialize b
```

Comment:
```suggestion
  // Launch an asynchronous kernel to initialize b.
```
Use `.` at the end of each sentence so the comments are parseable.

Reply:
Done in 372bb3b2d8462248c96972b5ea69396c776f006a.